torch_nets/models.py

```diff
@@ -35,55 +49,97 @@ def __init__(
         self,
         n_in: int = 61,
         n_out: int = 148,
-        neurons=128,
-        dropout=0.0,
+        n_layers: int = 5,
+        neurons: int = 128,
+        dropout: float = 0.0,
+        device: str = "cpu",
+        features_mean: Any = None,
+        features_std: Any = None,
+        outputs_mean: Any = None,
+        outputs_std: Any = None,
+        output_groups: Any = None,
     ):
-        """Build ``ANN``."""
-        super().__init__()
-        self.linear1 = Linear(n_in, neurons)
-        self.linear2 = Linear(neurons, neurons)
-        self.linear3 = Linear(neurons, neurons)
-        self.linear4 = Linear(neurons, neurons)
-        self.linear5 = Linear(neurons, n_out)
-
-        self.lin_drop1 = Dropout(dropout)
-        self.lin_drop2 = Dropout(dropout)
-        self.lin_drop3 = Dropout(dropout)
-        self.lin_drop4 = Dropout(dropout)
-
-    def forward(self, batch):
-        """Pass ``batch`` through the model.
+        """Initialize the ANN model."""
+        dims: List[int] = [n_in] + [neurons] * (n_layers - 1) + [n_out]
+        layers: List[Any] = []
```

Comment:
I've removed the type annotations here

Reply:
Can `dims` not  be typed `list[int]` as it was?\
Is it possible to make `layers` a `list[nn.Module]` or `list[Union[nn....]]`?

Reply:
It can but there are no type annotations of local variables elsewhere, so I thought it's unnecessary to specifically add `list[int]` and `list[nn.Module]` here.

Reply:
Usually if declaring it is good practice to specify type to be more constrained as mypy will try to infer it otherwise.
Given it passes and the variables are internal and passed up to Torch I think this is OK for now.